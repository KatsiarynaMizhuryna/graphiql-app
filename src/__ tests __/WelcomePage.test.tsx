import React, { ReactNode } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '@/app/page';
import Link from 'next/link';

interface LinkButtonProps {
  children: ReactNode;
  href: string;
}

jest.mock('@/components/welcomePage/mainContent/MainContent', () => ({
  MainContent: () => <div>MainContent</div>
}));
jest.mock('@/components/welcomePage/welcomeContent/WelcomeContent', () => ({
  WelcomeContent: () => <div>WelcomeContent</div>
}));

// jest.mock('@/components/welcomePage/mainContent/MainContent', () => ({
//   MainContent: ({ userName }: { userName: string }) => (
//     <div>MainContent</div>
//   ),
// }));

describe('Home component', () => {
  it('renders WelcomeContent when user is not logged in', () => {
    render(<Home />);
    expect(screen.getByText('WelcomeContent')).toBeInTheDocument();
  });

  // it('renders MainContent when user is logged in', () => {
  //   jest.spyOn(React, 'useState').mockReturnValue([true, jest.fn()]);

  //   render(<Home />);
  //   expect(screen.getByText('MainContent')).toBeInTheDocument();
  // });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});

jest.mock('@/ui/LinkButton', () => ({
  LinkButton: ({ children, href }: LinkButtonProps) => (
    <Link href={href}>{children}</Link>
  )
}));
