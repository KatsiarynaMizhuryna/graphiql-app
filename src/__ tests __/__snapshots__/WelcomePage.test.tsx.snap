// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="flex-grow container mx-auto flex flex-col items-center justify-around px-4 sm:px-8 md:px-12 lg:px-16 xl:px-20"
  >
    <div>
      WelcomeContent
    </div>
  </main>
</DocumentFragment>
`;
